MathLib/ODE/OdeSolverTypes.h

```diff
@@ -0,0 +1,67 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef MATHLIB_ODE_ODESOLVERTYPES_H
+#define MATHLIB_ODE_ODESOLVERTYPES_H
+
+#include <Eigen/Core>
+
+namespace MathLib
+{
+template <int M, int N>
+using MappedMatrix = Eigen::Map<Eigen::Matrix<double, M, N, Eigen::ColMajor>>;
+
+template <int M, int N>
+using MappedConstMatrix =
+    Eigen::Map<const Eigen::Matrix<double, M, N, Eigen::ColMajor>>;
+
+template <int N>
+using MappedVector = MappedMatrix<N, 1>;
+
+template <int N>
+using MappedConstVector = MappedConstMatrix<N, 1>;
+
+// y read only
+// ydot is written in the function to be called.
+template <unsigned N, typename... FunctionArguments>
+using Function = bool (*)(const double t,
+                          MappedConstVector<N> const y,
+                          MappedVector<N> ydot,
```

Comment:
```
	DBUG("sizes: map %lu, double* %lu, int %lu",
		 sizeof(Eigen::Map<Eigen::Matrix<double, 20, 20, Eigen::ColMajor>>),
		 sizeof(double*), sizeof(int));
```
says Eigen::Map is 16 bytes, i.e. probably one double* and two ints or so. Anyway, I'll change to pass by reference.